line_parser.c

```diff
@@ -71,14 +74,14 @@ DataInstruction* parse_data_instruction(char *line) {
     return instruction;
 }
 
-char * parse_data_line(char *line) {
-    IC++;
+char * parse_instruction_line(char *line) {
+    IC++; /* Why? if the 1st line of the assembly file is an instruction line, then we want to assemble it into cell 100 */
     DataInstruction *instruction = parse_data_instruction(line);
     printf("Label: %s|\n", instruction->label);
     printf("Opcode: %s|\n", instruction->opcode);
     printf("Operand 1: %s|\n", instruction->operand_1);
     printf("Operand 2: %s|\n", instruction->operand_2);
-    parse_instruction(instruction);
+    parse_instruction(instruction); /* Maybe convert_instruction (to machine language) would be a better name? */
```

Comment:
maybe, but naming isnt important

Reply:
It's just less confusing when reading the code, but I'm not gonna fight for that :-D